scripts/run_precommit_generation_scripts.py

```diff
@@ -0,0 +1,62 @@
+"""
+Script to run all "generation" scripts that require a python environment in sequence using
+the same environment. This avoids multiple virtual environment creations in pre-commit hooks.
+"""
+
+import importlib.util
+import sys
+from concurrent.futures import ProcessPoolExecutor
+from pathlib import Path
+from types import ModuleType
+
+
+def import_script(script_path: Path) -> ModuleType:
+    """Import a Python file as a module."""
+    spec = importlib.util.spec_from_file_location("module.name", str(script_path))
+    if spec is None:
+        raise ImportError(f"Could not load spec for {script_path}")
+
+    module = importlib.util.module_from_spec(spec)
+    if spec.loader is None:
+        raise ImportError(f"Could not load module for {script_path}")
+
+    spec.loader.exec_module(module)
+    return module
+
+
+def run_generator(script_name: str) -> None:
+    print(f"Running {script_name}...")
+
+    script_path = Path(__file__).parent / script_name
+
+    original_argv = sys.argv.copy()
+
+    try:
+        sys.argv = [str(script_path)]
```

Comment:
Is this `sys.argv` modification necessary? I don't think any of the generation scripts depend on it.